.github/workflows/ts-tests.yml

```diff
@@ -0,0 +1,88 @@
+# This workflow will install Typescript/React dependencies and run tests with a single version of Node
+
+name: ts-tests
+
+on:
+  push:
+    branches:
+      - trunk
+  pull_request:
+    branches:
+      - '**'
+
+# Because of https://github.com/actions/runner/issues/353, we need to explicitly set the shell to bash to get sane defaults (-o pipefail).
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  code-style:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node 14.15
+        id: setup
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.15.4
+      - name: Cache dependencies
+        id: cache-deps
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ts-tests-code-style-v0-${{ runner.os }}-${{ steps.setup.outputs.node-version }}-${{ hashFiles('.github/workflows/ts-tests.yml') }}
+      - name: Install dependencies
+        run: npm --prefix src/ts ci
+      - name: Verify code style
+        run: npm --prefix src/ts run lint-ci
+
+  tests:
+    runs-on: ubuntu-latest
+    needs: code-style
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node 14.15
+        id: setup
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.15.4
+      - name: Cache dependencies
+        id: cache-deps
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ts-tests-code-style-v0-${{ runner.os }}-${{ steps.setup.outputs.node-version }}-${{ hashFiles('.github/workflows/ts-tests.yml') }}
```

Comment:
If this is going to use the same deps as the code-style section, you might want to consider changing the key to be just `ts-tests-v0-...`